pkg/version/version.go

```diff
@@ -0,0 +1,40 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package version
+
+import (
+	"fmt"
+	"runtime"
+
+	"github.com/submariner-io/admiral/pkg/log"
+)
+
+var (
+	version       = "devel"
```

Comment:
Previously we used "not-compiled-properly" as the default, giving a clue about
issue in the build. Now we always show "devel" ignoring the VERSION environment
variable during build time.

To verify this change, try to build submariner with different VERSION values and
check the logs when the various components start to ensure that your change actually
does what it should.

Reply:
I'm trying to build submariner on my premises. but it failed. Am I missing something？

```shell
root@cyclinder3:~/submariner# make bin/%/submariner-gateway
docker network create  -d bridge kind
Error response from daemon: network with name kind already exists
make: [Makefile.dapper:57: bin/%/submariner-gateway] Error 1 (ignored)
./.dapper   -- make  bin/%/submariner-gateway
[+] Building 2.0s (8/8) FINISHED
 => [internal] load .dockerignore                                                                                                                            0.8s
 => => transferring context: 2B                                                                                                                              0.8s
 => [internal] load build definition from Dockerfile.dapper                                                                                                  0.8s
 => => transferring dockerfile: 629B                                                                                                                         0.8s
 => [internal] load metadata for quay.io/submariner/shipyard-dapper-base:devel                                                                               1.1s
 => [1/4] FROM quay.io/submariner/shipyard-dapper-base:devel@sha256:529d23ec7d03d1561b8222593d4f71f83ef63e9dc308ece7b4b40332aac434ce                         0.0s
 => CACHED [2/4] WORKDIR /go/src/github.com/submariner-io/submariner                                                                                         0.0s
 => CACHED [3/4] RUN git config --global --add safe.directory /go/src/github.com/submariner-io/submariner                                                    0.0s
 => CACHED [4/4] RUN git config --global --add safe.directory /go/src/github.com/submariner-io/submariner/ovn-kubernetes                                     0.0s
 => exporting to image                                                                                                                                       0.0s
 => => exporting layers                                                                                                                                      0.0s
 => => writing image sha256:77db9eab87dec54ea1dd4c1daa81210cff9572d66605b8a2dd0f9fd6ecf2eb43                                                                 0.0s
 => => naming to docker.io/library/submariner:version                                                                                                        0.0s
Checking if Makefile.dapper is up to date
Checking if Dockerfile.dapper is up to date
GOARCH=bin/%/ /opt/shipyard/scripts/compile.sh bin/%/submariner-gateway .
Running with environment variables:
BUILD_DEBUG=''
BUILD_UPX=true
LDFLAGS=-X\ github.com/submariner-io/submariner/pkg/version/version.Version=\"devel\"\ -X\ github.com/submariner-io/submariner/pkg/version/version.gitCommitHash=d18da3d16e23346e16c3998ea7666040232b3f1c\ -X\ github.com/submariner-io/submariner/pkg/version/version.gitCommitDate=2023-06-26T10:02:45+08:00
[02:07:49.583] [dir=submariner]$ mkdir -p bin/%
Building 'bin/%/submariner-gateway' (LDFLAGS: '-X github.com/submariner-io/submariner/pkg/version/version.Version="devel" -X github.com/submariner-io/submariner/pkg/version/version.gitCommitHash=d18da3d16e23346e16c3998ea7666040232b3f1c -X github.com/submariner-io/submariner/pkg/version/version.gitCommitDate=2023-06-26T10:02:45+08:00')
[02:07:49.592] [dir=submariner]$ LDFLAGS=-s -w -X github.com/submariner-io/submariner/pkg/version/version.Version="devel" -X github.com/submariner-io/submariner/pkg/version/version.gitCommitHash=d18da3d16e23346e16c3998ea7666040232b3f1c -X github.com/submariner-io/submariner/pkg/version/version.gitCommitDate=2023-06-26T10:02:45+08:00
[02:07:49.600] [dir=submariner]$ CGO_ENABLED=0 go build -trimpath -ldflags -s -w -X github.com/submariner-io/submariner/pkg/version/version.Version="devel" -X github.com/submariner-io/submariner/pkg/version/version.gitCommitHash=d18da3d16e23346e16c3998ea7666040232b3f1c -X github.com/submariner-io/submariner/pkg/version/version.gitCommitDate=2023-06-26T10:02:45+08:00 -o bin/%/submariner-gateway .
go: unsupported GOOS/GOARCH pair linux/bin/%/
make: *** [Makefile:68: bin/%/submariner-gateway] Error 2
make: *** [Makefile.dapper:58: bin/%/submariner-gateway] Error 2
root@cyclinder3:~/submariner# uname -a
Linux cyclinder3 5.15.0-75-generic #82-Ubuntu SMP Tue Jun 6 23:10:23 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
```